src/core/control/Control.cpp

```diff
@@ -1779,7 +1779,28 @@ void Control::undoRedoPageChanged(PageRef page) {
     }
 }
 
-void Control::selectTool(ToolType type) {
+void Control::handleSelectToolAction(ToolType type) {
```

Comment:
Based on the way this is implemented, I don't think this method needs to be renamed.